### PR TITLE
Obsolete methods that accepted a func without a cancellation token

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

**Comment by andreasohlund** [comment]
>ConfigureCriticalErrorAction.DefineCriticalErrorAction

Good catch, fixed!

>CriticalErrorContext.Stop (a property though, this one is odd)

Properties can't differ by just return type so this one can't be obsoleted?

File: src/NServiceBus.Core/obsoletes-v8.cs [hunk 1]
```diff
@@ -1518,4 +1518,141 @@ public partial class MessageContext
...
+    using Pipeline;
+
+    public static partial class ReceivePipelineConfigExtensions
+    {
+        [ObsoleteEx(
+            Message = "Use the overload that accepts a callback with a cancellation token.",
+            TreatAsErrorFromVersion = "8",
+            RemoveInVersion = "9")]
+        public static void OnReceivePipelineCompleted(this PipelineSettings pipelineSettings, Func<ReceivePipelineCompleted, Task> subscription)
+        {
```
[the remark this answers is not in the document]
Comment (andreasohlund) [reply src/NServiceBus.Core/obsoletes-v8.cs:1619]: I made all of them throw just in case

**Review by adamralph — COMMENTED** [review]
In some cases we describe the argument as a "callback":

> Use the overload that accepts a callback with a cancellation token

and in other cases we refer to it by its type (`Func`):

> Use the overload that accepts a Func with a cancellation token.

Should we be consistent and refer to it as the "callback" in all cases?

BTW - personally I'd prefer "delegate" to "callback" but not strongly.

**Comment by andreasohlund** [comment]
>Should we be consistent and refer to it as the "callback" in all cases?

👍  I've changed them all to "delegate"

**Review by DavidBoike — APPROVED** [review]

**Comment by DavidBoike** [comment]
Approved but looks like APIApprover needs to be rerun.

**Pull request merged**